README.md

```diff
@@ -1,117 +1,170 @@
+
+* [Entrypoint.](/entrypoint) This service is the interface and connects the previous microservices together. It is the entrypoint for the users to access the services.
+
+## Prerequisites
+
+You may want to create separate virtual environments for each of the microservices or you can create one for all of them. We advise to use virtual environments if you are developing multiple projects with Python, due to clashing of dependencies between projects. (Suppose one project only supports numpy < 1.0 and the other needs numpy=1.5).
+
+To create a virtual environment navigate to the desired directory (usually the main folder of the project) and write
+```bash
+python -m venv venv
+```
+To activate this virtual environment navigate into venv/Scripts and then execute `activate`. To deactivate a virtual environment execute `deactivate`.
 
 You can see that your virtual environment is being used if you see __(venv)__ before the command line.
 
-### DOCUMENT RETRIEVAL MICROSERVICE
+Each microservice must be run separately. Each service can be used for themself or one can employ the `entrypoint` microservice that connects all of the microservices together.
+
+What follows is a short description of how to run each microservice. A more detailed description of the microservice can be found in their designated folders.
+
+### Text Embeddings Microservice
+
+Currently you are able to run only one version of the text embedding so that it will be connected to the main component. But later you will be able to connect more.
+
+* Activate virtual environment if you wish to do so
+* Navigate into `text_embeddings` folder
+* Execute
+  ```bash
+  pip install -r requirements.txt
+  ```
+* Place a copy of your `word2vec` or `fasttext` word embeddings in the [data/embeddings](text_embeddings/data/embeddings) folder
+* Execute
+  ```bash
+  # linux or mac
+  python -m text_embedding.main start \
+         -H localhost \
+         -p 4001 \
+         -mp (path to the model) \
+         -ml (language of the model)
+
+  # windows
+  python -m text_embedding.main start -H localhost -p 4001 -mp (path to the model) -ml (language of the model)
+  ```
+
+
+### Document Retrieval Microservice
 
 * Activate virtual environment if you wish to do so
 * Navigate into `document_retrieval` folder
-* run `pip install -r requirements.txt`
+* Execute
+  ```bash
+  pip install -r requirements.txt
+  ```
 * Navigate into documentRetrieval/config folder
 * Create `.env` file and inside define the following variables:
-```
-PROD_PG_DATABASE={name_of_the_db}
-PROD_PG_PASSWORD={password}
+  ```bash
+  PROD_PG_DATABASE={name_of_the_db}
+  PROD_PG_PASSWORD={password}
 
-DEV_PG_DATABASE={name_of_the_db}
-DEV_PG_PASSWORD={pasword}
-```
+  DEV_PG_DATABASE={name_of_the_db}
+  DEV_PG_PASSWORD={pasword}
+  ```
 * Navigate into `documentRetrieval` folder and run the service with:
```

Comment:
The correct way to run the service is to navigate to document_retrieval and then run 
python -m documentRetrieval.main start -e development -p 5000
for other services in similar way

Reply:
Thanks for the comment! 

Can I also rename the `documentRetrieval` folder to `microservice` to be consistent with the rest microservices?